tests/unit/repo/test_package_repository.py

```diff
@@ -23,18 +23,42 @@
     PackageRepositoryAptPPA,
 )
 
-
-def test_apt_name():
-    repo = PackageRepositoryApt(
+# region Test data and fixtures
+BASIC_PPA_MARSHALLED = {"type": "apt", "ppa": "test/foo"}
+BASIC_APT_MARSHALLED = {
+    "architectures": ["amd64", "i386"],
+    "components": ["main", "multiverse"],
+    "formats": ["deb", "deb-src"],
+    "key-id": "A" * 40,
+    "key-server": "keyserver.ubuntu.com",
+    "name": "test-name",
+    "suites": ["xenial", "xenial-updates"],
+    "type": "apt",
+    "url": "http://archive.ubuntu.com/ubuntu",
+}
+
+
+@pytest.fixture
+def apt_repository():
+    yield PackageRepositoryApt(
```

Comment:
a minor curiosity question: why `yield` instead of `return` here?

Reply:
I got into the habit when I was writing a lot of fixtures that needed teardown, and now I have muscle memory for it. If you (or anyone else on the team) don't like it I'll try to break the habit.